jekyll/_cci2/server/installation/migrate-from-server-2-to-server-4.adoc

```diff
@@ -0,0 +1,109 @@
+
+toc::[]
+
+## Prerequisites
+
+. Your current CircleCI Server installation is 2.19.
+. You have taken a backup of the 2.19 instance. If you are using external datastores, they need to be backed up separately.
+. You have a new CircleCI Server 4.0 xref:phase-1-prerequisites.adoc[installation].
+. You have successfully run https://support.circleci.com/hc/en-us/articles/360011235534-Using-realitycheck-to-validate-your-CircleCI-installation[realitycheck] with contexts before starting.
+. The migration script must be run from a machine with:
+- `kubectl` configured for the server 4.0 instance
+- `ssh` access to the 2.19 services box
+
+### External Datastores Only
+. Backups have been taken of all external data stores.
+. Postgres has been updated to version 12.
+
+### Internal Datastore Only
+. You have taken a backup of the 2.19 instance.
+. You have successfully run https://support.circleci.com/hc/en-us/articles/360011235534-Using-realitycheck-to-validate-your-CircleCI-installation[realitycheck] on the new server 4.0 instance with contexts prior to starting.
+
+## Migration
+
+WARNING: Migrating to server v4.0 will shut down your v2.19 application. Your v2.19 application will not be restarted, although you may manually restart it using the KOTS Admin Console.
+
+WARNING: Starting the migration process will cause downtime. It is recommended you schedule a maintenance window.
+
+WARNING: Running server 2.19 and server 4.0 at the same time can cause issues with your 2.19 build data. Server 2.19 should NOT be restarted if server 4.0 is running.
+
+### Step 1 - Clone the repository and run the migration script
+The instructions below will clone the repository containing the server v2.19.x to server v4.0 migration script.
+
+The migration script will:
+
+* Stop your v2.19.x application.
+* Perform preflight checks to confirm namespace and datastores for 2.19.x. 
+* Create a tarball of your v2.19.x application's PostgreSQL and Mongo databases.
+* Archive existing application data for Vault and CircleCI encryption/signing keys.
+* Export the 2.19.x tarball to your v4.0 installation. Exported data stores are stored in a directory named `circleci_export`, located relative to wherever the migration script is run from. This can be useful for debugging purposes.
+* Perform preflight checks to confirm namespace and datastores for 4.0 instance. 
+* Scale v4.0 application deployments down to zero.
+* Import the data from the previously exported tarball to your new v4.0 instance.
+* Scale v4.0 application deployments up to one.
+
+NOTE: If you have externalized services, you can run `bash migrate.sh --server4 -v -p -m`. These `-v -p -m` flags will skip the migration of Vault, Postgres, and Mongo, respectively. Skipping all three will copy your keys from `/data/circle/circleci-encryption-keys` on the v2.19.x services machine, allowing you to `cat` these files and upload their contents to the 4.0 configuration page.
```

Comment:
At which point can they do this? Would be good to get this one in sequence in the steps below

Reply:
they can use the flags in the first step where we run `migrate.sh`
i've moved the NOTE to be in the step